File: scripts/algorithms/N/Number of Different Integers in a String/Number of Different Integers in a String.java
// Runtime: 22 ms (Top 16.67%) | Memory: 44.5 MB (Top 21.67%)
class Solution {
    public int numDifferentIntegers(String word) {
        String[] arr = word.replaceAll("[a-zA-Z]", " ").split("\\s+");
        Set<String> set = new HashSet<String>();

        for (String str : arr) {
            if (!str.isEmpty())
                set.add(String.valueOf(str.replaceAll("^0*","")));
        }

        return set.size();
    }
}

File: scripts/algorithms/N/Number of Different Integers in a String/Number of Different Integers in a String.py
// Runtime: 34 ms (Top 85.83%) | Memory: 16.50 MB (Top 46.61%)

class Solution:
    def numDifferentIntegers(self, word: str) -> int:
        word = re.findall('(\d+)', word)
        nums = [int(i) for i in word]
        
        return len(set(nums))
